modules/core/src/main/java/org/apache/synapse/mediators/builtin/ValidateMediator.java

```diff
@@ -220,7 +240,24 @@ public boolean mediate(MessageContext synCtx) {
                         } else {
                             handleException("Can not find valid JSON Schema content", synCtx);
                         }
-                        jsonSchema = jsonSchemaFactory.getJsonSchema(jsonSchemaNode);
+                        cachedJsonSchema = jsonSchemaFactory.getJsonSchema(jsonSchemaNode);
+                        
+                        /*
+                         * Initially adds the cached schema to the map if it's
+                         * not available
+                         */
+                        if (!cachedJsonSchemaMap.containsKey(cachedJsonSchemaKey.toString())) {
+                            cachedJsonSchemaMap.put(cachedJsonSchemaKey.toString(), cachedJsonSchema);
+                            /*
+                             * Removes the existing cached schema and adds the
+                             * new cached schema This is used when editing a
+                             * registry resource or when the cache expires
+                             */
+                        } else if (cachedJsonSchemaMap.containsKey(cachedJsonSchemaKey.toString())) {
+
```

Comment:
unnecessary new line, Shall we remove it?

Reply:
done